backends/p4tools/modules/testgen/options.cpp

```diff
@@ -36,32 +41,79 @@ TestgenOptions::TestgenOptions()
             inputPacketOnly = true;
             return true;
         },
-        "Only produce the input packet for each test");
+        "Only produce the input packet for each test.");
 
     registerOption(
         "--max-tests", "maxTests",
         [this](const char* arg) {
-            maxTests = std::atoi(arg);
+            try {
+                maxTests = std::stoull(arg);
+            } catch (std::invalid_argument&) {
+                ::error("Invalid input value %1% for --max-tests. Expected integer.", arg);
+                return false;
+            }
             return true;
         },
-        "Sets the maximum number of tests to be generated");
+        "Sets the maximum number of tests to be generated [default: 1]. Setting the value to <=0 "
+        "will generate tests until no more paths can be found.");
```

Comment:
I am not sure if a negative number of tests makes sense. It's a bit counterintuitive, I think. It could be a mistake. If we want to exhaustively generate tests, I think it should be a specific stop metric, along with `MAX_STATEMENT_COVERAGE`.

More generally, here's a suggestion: every time the user inputs an arg that does not make much sense, I believe we should default it to something that won't take too long to terminate and produce some tests.

Reply:
I made it such that an error is thrown if the input is negative. I still think having 0 as value to generate infinite tests is an intuitive input since it is used in a fair number of applications. It can also be easily combined with other stop metrics.

Reply:
The only issue I see with that is non-termination. I'm not sure if it's explicit to the user it can happen. That's why I was suggesting enforcing a bounded model; if not, then use an explicit flag.

Reply:
In this case, non-termination is an explicit user choice (0 tests, which is not a default) and the stop metric is exhaustion of all available branches. P4Testgen will generate all the unique tests for that particular program, then stop. 

The only case where non-termination is possible is a unconditional parser-loop. There the program graph is not a DAG anymore. P4Testgen currently does not handle those correctly. Ideally, we would bound these in some form.